vllm/config.py

```diff
@@ -4075,3 +4075,16 @@ def assert_hashable(text):
         f"vLLM tried to hash some configs that may have Python objects ids "
         f"in them. This is a bug, please file an issue. "
         f"Text being hashed: {text}")
+
+
+T = TypeVar("T")
+
+
+def get_layers_from_vllm_config(vllm_config: VllmConfig,
+                                layer_type: type[T]) -> dict[str, T]:
+    return {
+        layer_name: layer
+        for layer_name, layer in
+        vllm_config.compilation_config.static_forward_context.items()
+        if isinstance(layer, layer_type)
+    }
```

Comment:
Are there any caveats to the static forward context? Can we always use this or only sometimes? (I'm not sure) Could you add a comment if there's anything folks should be careful of?

Reply:
I've updated the comment to show that there may be different types of layers in forward context.